pkg/controller/crdmirroring/crdmirroring_handlers.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crdmirroring
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/controller/crdmirroring/types"
+)
+
+func (c *Controller) onNewCRDAdd(obj interface{}) {
+	CRD := obj.(metav1.Object)
+
+	if CRD == nil {
+		utilruntime.HandleError(fmt.Errorf("onNewCRDAdd() got unexpected type: %T", obj))
+		return
+	}
+
+	// Check if the legacy CRD mirroring current CRD exists.
+	if CRD.GetAnnotations()[types.ManagedBy] == types.ControllerName {
+		klog.Infof("Processing mirroring %s %s/%s CHECK event", c.mirroringCRDName, CRD.GetNamespace(), CRD.GetName())
+		c.queueCRD(obj)
+	}
+}
+
+func (c *Controller) onNewCRDUpdate(prevObj, obj interface{}) {
+	prevCRD := prevObj.(metav1.Object)
+	CRD := obj.(metav1.Object)
+
+	if CRD == nil || prevCRD == nil {
+		utilruntime.HandleError(fmt.Errorf("onNewCRDUpdate() got unexpected type: %T, %T", prevObj, obj))
+		return
+	}
+
+	if CRD.GetAnnotations()[types.ManagedBy] == types.ControllerName && prevCRD.GetAnnotations()[types.ManagedBy] == types.ControllerName {
```

Comment:
Is it necessary to check previous annotation?

Reply:
No, I'll delete it.